imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -0,0 +1,208 @@
+        cdf_file = write_cdf(dataset)
+        cdf_filepaths.append(cdf_file)
+    logger.info("L1A CDF files created")
+    return cdf_filepaths
+
+
+def decom_packets(packet_file: str):
+    """
+    Unpack and decode packets using CCSDS file and XTCE packet definitions.
+
+    Parameters
+    ----------
+    packet_file : str
+        Path to the CCSDS data packet file.
+
+    Returns
+    -------
+    list
+        List of all the unpacked data
+    """
+    # TODO: update path to use a combined packets xtce file
+    xtce_file = imap_module_directory / "hit/packet_definitions/P_HIT_HSKP.xml"
+    logger.info(f"Unpacking {packet_file} using xtce definitions in {xtce_file}")
+    unpacked_packets = decom.decom_packets(packet_file, xtce_file)
+    logger.info(f"{packet_file} unpacked")
+    return unpacked_packets
+
+
+def group_data(unpacked_data: list):
+    """Group data by apid.
+
+    Parameters
+    ----------
+    unpacked_data : list
+        packet list
+
+    Returns
+    -------
+    dict
+        grouped data by apid
+    """
+    logger.info("Grouping packet values for each apid")
+    grouped_data = utils.group_by_apid(unpacked_data)
+
+    # Create data classes for each packet
+    for apid in grouped_data.keys():
+        if apid == HitAPID.HIT_HSKP:
+            logger.info(f"Grouping housekeeping packets - APID: {apid}")
+            for i, packet in enumerate(grouped_data[apid]):
+                # convert data to data classes
+                grouped_data[apid][i] = Housekeeping(packet, "0.0", "hskp_sample.ccsds")
+        else:
+            raise RuntimeError(f"Encountered unexpected APID [{apid}]")
+
+    logger.info("Finished grouping packet data")
+    return grouped_data
+
+
+def create_datasets(data: dict, skip_keys=None):
+    """
+    Create a dataset for each APID in the data.
+
+    Parameters
+    ----------
+    data : dict
+        A single dictionary containing data for all instances of an APID.
+    skip_keys: list, Optional
+        Keys to skip in the metadata
+
+    Returns
+    -------
+    dict
+        A dictionary containing xr.Dataset for each APID. Each dataset in the
+        dictionary will be converted to a CDF.
+    """
+    logger.info("Creating datasets for HIT L1A data")
+    processed_data = {}
+    for apid, data_packets in data.items():
+        metadata_arrays = defaultdict(list)
+        for packet in data_packets:
+            # Add metadata to an array
+            for field in fields(packet):
+                field_name = field.name
+                field_value = getattr(packet, field_name)
+                print(f"{field_name}: {field_value}")
+                # convert key to lower case to match SPDF requirement
+                data_key = field_name.lower()
+                metadata_arrays[data_key].append(field_value)
+
+        # Create xarray data arrays for dependencies
+        epoch_time = xr.DataArray(
+            np.array(metadata_arrays["shcoarse"], dtype="datetime64[s]"),
```

Comment:
Just an FYI that SHCOARSE is an integer, so I think you will need to use the `calc_start_time()` function to get this value into a `datetime64[s]`, otherwise this is actually going to be showing 1970's unix epoch time values.

Reply:
Updated!